src/components/buttonSelectors/MultiButtonSelector.js

```diff
@@ -1,117 +1,159 @@
+      let vehicleID = dependencyID[1];
+      switch (dependencyID.length) {
+        case 2:
+          // Case 2 updates the vehicle form
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: idCode,
+              vehicleID: vehicleID,
+            });
+          break;
+        case 3:
+          // Case 3 updates the passenger form
+          let passengerID = dependencyID[2];
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: idCode,
+              vehicleID: vehicleID,
+              passengerID: passengerID,
+            });
+        case 4:
+          // Case 4 updates the nonmotorist form
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: idCode,
+              vehicleID: vehicleID,
+              nonmotoristID: dependencyID[3],
+            });
+        default:
+          break;
+      }
     }
+    submitFunction({ id, question: currId, selection: optionText });
+  };
 
-    const renderSingleButton = (option) => {
-        return (
-            <Button
-                key={option.id}
-                mr={4}
-                mb={4}
-                variant={selection === option.name ? 'solid': 'subtle'}
-                onPress={() => submitField(option.name, option.id)}
-            >
-                {option.name}
-            </Button>
-        )
-    }
+  /*
+    This function renders each of the buttons and sets their styling.
+  */
+  const renderSingleButton = (option) => {
+    return (
+      <Button
+        key={option.id}
+        mr={4}
+        mb={4}
+        variant={selection === option.name ? "solid" : "subtle"}
+        onPress={() => submitField(option.name, option.id)}
+      >
+        {option.name}
+      </Button>
+    );
+  };
 
-    const renderButtons = () => {
-        let res = data.answerOptions.map(option => (renderSingleButton(option)));
-        return (
-            <HStack flexWrap='wrap'>
-                {res}
-            </HStack>
-        );
-    }
+  /*
+    This function renders the buttons based on the answer options.
+  */
+  const renderButtons = () => {
+    let res = data.answerOptions.map((option) => renderSingleButton(option));
+    return <HStack flexWrap="wrap">{res}</HStack>;
+  };
 
-    const tooltip = () => {
-        return(<TooltipView toolTip={data.tooltip} helperImg={data.helperImg}/>)
-    }
+  // Renders tooltip component
+  const tooltip = () => {
+    return <TooltipView toolTip={data.tooltip} helperImg={data.helperImg} />;
+  };
 
-    var renderComponent = true;
-    if (data.questionDependency != undefined && props.response != null) {
-        renderComponent = dependencyParser(props.response, data, dependencyID)
-    }
+  let renderComponent = true;
+  // Check if the question has any dependencies to update
```

Comment:
this should be rephrased to "check if there are dependencies that need to be satisfied before the question can be rendered"